huaweicloud/services/acceptance/ddm/data_source_huaweicloud_ddm_instance_nodes_test.go

```diff
@@ -0,0 +1,88 @@
+package ddm
+
+import (
+	"fmt"
+	"strings"
+	"testing"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/resource"
+
+	"github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance"
+)
+
+func TestAccDatasourceDdmInstanceNodes_basic(t *testing.T) {
+	name := acceptance.RandomAccResourceName()
+	name = strings.ReplaceAll(name, "_", "-")
+	rName := "data.huaweicloud_ddm_instance_nodes.test"
+	dc := acceptance.InitDataSourceCheck(rName)
+
+	resource.ParallelTest(t, resource.TestCase{
+		PreCheck:          func() { acceptance.TestAccPreCheck(t) },
+		ProviderFactories: acceptance.TestAccProviderFactories,
+		Steps: []resource.TestStep{
+			{
+				Config: testAccDatasourceDdmInstanceNodes_basic(name),
+				Check: resource.ComposeTestCheckFunc(
+					dc.CheckResourceExists(),
+					resource.TestCheckResourceAttr(rName, "nodes.#", "2"),
+					resource.TestCheckResourceAttr(rName, "nodes.0.status", "normal"),
+				),
+			},
+		},
+	})
+}
+
+func testAccDatasourceDdmInstanceNodes_base(name string) string {
+	return fmt.Sprintf(`
+resource "huaweicloud_vpc" "test" {
+  name = "%[1]s"
+  cidr = "192.168.0.0/24"
+}
+
+resource "huaweicloud_vpc_subnet" "test" {
+  name       = "%[1]s"
+  cidr       = "192.168.0.0/24"
+  gateway_ip = "192.168.0.1"
+  vpc_id     = huaweicloud_vpc.test.id
+}
+
+resource "huaweicloud_networking_secgroup" "test" {
+  name = "%[1]s"
+}
+
+data "huaweicloud_availability_zones" "test" {}
+
+data "huaweicloud_ddm_engines" test {
+  version = "3.0.8.5"
+}
+
+data "huaweicloud_ddm_flavors" test {
+  engine_id = data.huaweicloud_ddm_engines.test.engines[0].id
+  cpu_arch  = "X86"
+}
+
+resource "huaweicloud_ddm_instance" "test" {
+  name              = "%[1]s"
+  flavor_id         = data.huaweicloud_ddm_flavors.test.flavors[0].id
+  node_num          = 2
+  engine_id         = data.huaweicloud_ddm_engines.test.engines[0].id
+  vpc_id            = huaweicloud_vpc.test.id
+  subnet_id         = huaweicloud_vpc_subnet.test.id
+  security_group_id = huaweicloud_networking_secgroup.test.id
+
+  availability_zones = [
+    data.huaweicloud_availability_zones.test.names[0]
+  ]
+}
+`, name)
+}
+
+func testAccDatasourceDdmInstanceNodes_basic(name string) string {
+	return fmt.Sprintf(`
+%s
+
+data "huaweicloud_ddm_instance_nodes" "test" {
+  instance_id = huaweicloud_ddm_instance.test.id
+}
+`, testAccDatasourceDdmInstanceNodes_base(name))
```

Comment:
Maybe you can use exist function `testDdmInstance_basic` in `resource_huaweicloud_ddm_instance_test.go`.